src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,112 @@
+package duke.parser;
```

Comment:
this file is super arcane! Would suggest being more declarative and breaking them up into smaller functions!